src/backend/metatypes.h

```diff
@@ -120,7 +46,35 @@ namespace snmalloc
      * bit packing does not conflict; see mem/remoteallocator.h
      */
     static constexpr address_t REMOTE_BACKEND_MARKER = 1 << 7;
+  };
+
+  /**
+   * Entry stored in the pagemap.  See docs/AddressSpace.md for the full
+   * MetaEntry lifecycle.
+   */
+  template<typename BackendMetadata>
+  class MetaEntry : public MetaEntryBase
```

Comment:
If PageMapEntry should be a derived class from MetaEntry, then it needs some way to update the underlying meta and remote_and_sizeclass fields.

Reply:
Also having the copy-constructor deleted and needing to implement the assignment operator is more awkward for inheritance.

Reply:
That is unchanged from the previous version.  I agree that this class isn't the easiest to extend, but since you're currently the only person with a use case for extending it then I'd rather that we had an orthogonal PR to address those concerns, after you've tried to use it so that we have something that definitely works and not something that I hope works.

Reply:
The copy assignment operator is used only by the pagemap.  The new version demonstrates a back end that extends this class and makes all constructors private / friends, so you can see exactly what you need to support for copying / construction in an extended version.

Reply:
If you want to subclass, then I should remove the `Pagemap::Entry`.  It's there as an example of what to do if you want a custom metadata entry, but it's not intended to be extensible.